File: leetcode/c++/junechallengeday1.cpp
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if(!root)
            return nullptr;
        
        // get the root of respective subtrees
        TreeNode* left = invertTree(root->left);
        TreeNode* right = invertTree(root->right);
        
        // swap the subtrees
        root->left = right;
        root->right = left;
        
        return root;
    }
};

File: leetcode/python/junechallengeday1.py
class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        def bst(root):
            if root.left==None and root.right==None:
                return
            t=root.left
            root.left=root.right
            root.right=t
            if root.left!=None:
                bst(root.left)
            if root.right!=None:
                bst(root.right)
        if root!=None:
            bst(root)
        return  root   
